### DISCUSSION TITLE
treat query/**.scm as query

### DISCUSSION THREAD

**Discussion by Freed-Wu** [nvim-treesitter/nvim-treesitter#6080] [Ideas]
According to <https://tree-sitter.github.io/tree-sitter/syntax-highlighting#queries>:

> Syntax highlighting is controlled by three different types of query files that are usually included in the queries folder. The default names for the query files use the .scm file. We chose this extension because it commonly used for files written in Scheme, a popular dialect of Lisp, and these query files use a Lisp-like syntax.

So how about add a `ftdetect/query.vim` to higlight all query scm to query?

```vim
augroup XXX
  autocmd!
  autocmd BufNewFile,BufRead */queries/*/*.scm setfiletype query
augroup END
```

**Comment by theHamsta** [comment]
nvim-treesitter has removed all `ftdetect` files. These should be managed by upstream vim/vim or neovim/neovim. For Neovim, the following rule has already been added https://github.com/neovim/neovim/blob/607606e3bb7a435ac5a0eba2eb4abde8eff774bb/runtime/lua/vim/filetype.lua#L1913

**Reply by Freed-Wu** [reply]
I found why it cannot work for me. <https://github.com/sheerun/vim-polyglot/blob/master/autoload/polyglot/init.vim#L627>:

```vim
if !has_key(g:polyglot_is_disabled, 'scheme')
  au BufNewFile,BufRead *.scm,*.ss setf scheme
endif
```

**Reply by Freed-Wu** [reply]
And <https://github.com/sheerun/vim-polyglot/blob/master/extras/filetype.vim#L1724>:

```vim
" Scheme
au BufNewFile,BufRead *.scm,*.ss,*.sld,*.rkt,*.rktd,*.rktl 	setf scheme
```

**Reply by clason** [reply]
Yes, avoid polyglot like the plague. Neovim already has (correct) built-in detection for query files, you don't need to (and shouldn't do) anything.
